Source/monster.cpp

```diff
@@ -2713,19 +2668,15 @@ void SneakAi(int monsterId)
 	if (monster.mode != MonsterMode::Stand) {
 		return;
 	}
-	int mx = monster.position.tile.x;
-	int my = monster.position.tile.y;
-	if (dLight[mx][my] == LightsMax) {
+	if (dLight[monster.position.tile.x][monster.position.tile.y] == LightsMax) {
 		return;
 	}
-	mx -= monster.enemyPosition.x;
-	my -= monster.enemyPosition.y;
 
 	int dist = 5 - monster.intelligence;

```

Comment:
Can this value ever be negative?

Reply:
From what I can see, `intelligence` is between 0 and 3. So that's always positive.

Reply:
No monster has more then 3 in intelligence, so no.